app/helpers/requests_helper.rb

```diff
@@ -45,6 +45,15 @@ def label_for_item_selector_holding(holding)
     end
   end
 
+  def queue_length_display(item)
+    return '' if item.status == 'Available'
+
+    queue_display = item.checked_out? ? "Checked out - Due #{item.due_date} | " : "#{item.status} | "
+    return "#{queue_display}No waitlist" if item.queue_length.zero?
+
+    "#{queue_display}There is a waitlist ahead of your request"
```

Comment:
How about:

```suggestion
    item_status = item.checked_out? ? "Checked out - Due #{item.due_date}" : item.status
    wait_list = item.queue_length.zero? ? "No waitlist" : "There is a waitlist ahead of your request"
    
    "#{item_status} | #{wait_list}"
```

I think it's a little easier to read if we separate figuring out the values from doing the formatting.